src/cache.rs

```diff
@@ -74,22 +94,38 @@ impl Cache {
             }
         }
         let client = builder.build().unwrap_or_else(|_| Client::new());
-        let mut resp = client.get(&self.url).send()?;
+        let mut resp = client.get(&self.source).send()?;
         let mut buf: Vec<u8> = vec![];
         let bytes_downloaded = resp.copy_to(&mut buf)?;
         debug!("{} bytes downloaded", bytes_downloaded);
         Ok(buf)
     }
 
+    /// Download the archive from the network or load it from a file.
+    #[cfg(feature = "networking")]
+    fn load(&self) -> Result<Vec<u8>, TealdeerError> {
+        if self.source.starts_with("http") {
```

Comment:
Nitpicking, maybe something less stringly-typed? 😄 

```rust
pub enum Source {
    File(PathBuf),
    #[cfg(feature = "networking")]
    URL(String),
}
```

Reply:
Hm, good point. Then we can move the parsing up in the hierarchy and convert the input string right after argument parsing :slightly_smiling_face: